hyperspy/io_plugins/fei.py

```diff
@@ -580,13 +619,53 @@ def ser_reader(filename, objects=None, verbose=False, *args, **kwds):
         'mapping': mapping}
     return dictionary
 
-
-def get_mode(mode):
+def guess_units_from_mode(objects_dict, header, verbose=False):
+    # assuming that for an image stack, the UnitsLength of the "3rd" dimension
+    # is 0
+    isImageStack = (header['Dim-1_UnitsLength'][0] == 0)
```

Comment:
There are ser files that don't contain "Dim" information in the header, so this line may fail on them. See #835